src/TLuaInterpreter.cpp

```diff
@@ -3989,18 +3992,62 @@ int TLuaInterpreter::connectToServer( lua_State *L )
 
     if( ! lua_isnumber( L, 2 ) )
     {
-        lua_pushstring( L, "connectToServer: wrong argument type" );
-        lua_error( L );
-        return 1;
+        lua_pushstring( L, QString(" connectToServer: argument #2 expects type int, %1 was given.").arg( lua_typename(L,2) ).toUtf8().constData() );
+        return lua_error( L );
     }
     else
     {
         port = lua_tonumber( L, 2 );
+
+        if( port > 65535 || port < 0 )
+        {
+            lua_pushstring( L, " connectToServer: argument #2 is invalid, a valid port number is required.");
+            return lua_error(L);
+        }
     }
+
+    // Optional argument to save this new connection to disk for this profile.
+    if( !lua_isnoneornil(L,3) && !lua_isboolean(L,3) ) {
+        lua_pushstring( L, QString(" connectToServer: argument #3 expects boolean type, %1 was given.").arg( lua_typename(L,3) ).toUtf8().constData() );
+        return lua_error(L);
+    }
+    if( ! lua_isnoneornil(L,3) && lua_isboolean(L,3)) {
+        saveToProfile = lua_toboolean(L,3);
+        if( saveToProfile ) {
+            qDebug() << "TLuaInterpreter::connectToProfile(" << url.c_str() << ", " << port << ", " << saveToProfile << ")";
+
+            const QString filePath("%1/.config/mudlet/profiles/%2/%3");
```

Comment:
Unfortunately it won't help you at the moment but in another PR #969 there is a `Host` class `(void) writeProfileData(const QString &`\<item>`, const QString &` \<data>`)` method (derived from `(void) dlgConnectionProfiles::writeProfileData( QString `\<profile>`, QString `\<item>`, QString `\<what>` )`) waiting to go into the code-base that would almost provide the functionality you are using here...

You are only testing for the ability to open the file for writing but there are other failure modes that will not be detected as the code currently stands - if I push a `QPair<bool, QString> Host::writeProfileData(const QString &`\<item>`, const QString &` \<data>`)` out to this PR on your repository - so that the `first` part of the `QPair` is true on success and the `second` part is the `errorString()` message on failure - would that be of use here (and I can revise my PR to accommodate the change as well)...?